src/ristretto.rs

```diff
@@ -296,7 +296,7 @@ impl CompressedRistretto {
         // t == ((1+as²) sqrt(4s²/(ad(1+as²)² - (1-as²)²)))/(1-as²)
         let t = &x * &y;
 
-        if ok.unwrap_u8() == 0u8 || t.is_negative().unwrap_u8() == 1u8 || y.is_zero().unwrap_u8() == 1u8 {
+        if (ok.unwrap_u8() == 0u8) | (t.is_negative().unwrap_u8() == 1u8) | (y.is_zero().unwrap_u8() == 1u8) {
```

Comment:
Given that `Choice` implements `BitXor` and `Not`, I think this could be written more simply as:
```suggestion
        if ((!ok | t.is_negative() | y.is_zero()).unwrap_u8() == 1u8) {
```